powerline/segments/i3wm.py

```diff
@@ -80,23 +83,26 @@ def workspace(pl, segment_info, workspace=None, strip=False):
 		try:
 			w = next((
 				w for w in get_i3_connection().get_workspaces()
-				if w['name'] == workspace
+				if w.name == workspace
 			))
 		except StopIteration:
 			return None
 	elif segment_info.get('workspace'):
-		w = segment_info['workspace']
+		w = Dummy()
+		info = segment_info['workspace']
+		for k in info:
+			setattr(w, k, info[k])
```

Comment:
kind of an eyesore but what can you do... I tried `setattr` on the dict itself, but since it's a CPython object that doesn't work.

Reply:
Personally, I use the following to obtain a new dummy workspace, perhaps something similar could work here?
```
import copy
def get_next_ws(ws, outputs):
    names = [w.name for w in ws] 
    for i in range(1, 100):
        if not str(i) in names:
            res_ls = []
            res = copy.deepcopy(ws[0])
            res.num = -5
            res.name = str(i)
            res.urgent = False
            res.focused = False
            res.visible = False
            for o in outputs:
                r2 = copy.deepcopy(res)
                r2.output = o 
                res_ls += [r2]
            return res_ls
    return []
```

Reply:
Hm, wondering whether there is a use case where we would like that segment to work *without* a connection to i3, because it seems like the Dummy class currently allows that where your solution wouldn't.

Reply:
You could probably update my solution to also add dummy values for the output (and perhaps the other things that a workspace object should have); then you wouldn't need a connection anymore (but the code would become even more ugly).